libpod/container_internal.go

```diff
@@ -496,6 +496,11 @@ func (c *Container) setupStorage(ctx context.Context) error {
 		}
 	}
 	if containerInfoErr != nil {
+		if errors.Is(containerInfoErr, storage.ErrDuplicateName) {
+			if _, err := c.runtime.LookupContainer(c.config.Name); err != nil {
+				return fmt.Errorf("creating container storage: %w by an external entity", containerInfoErr)
```

Comment:
You should make sure the error is define.ErrNoSuchCtr

If you want to give the user a hint to remove you could state: 

"External container in container/storage must be removed first: %w"

Reply:
`Error: creating container storage: the container name \"qwi9mpytct\" is already in use by a3ffce20528fed3b9e1247c5367ce1059484f1f1d4e40f5dcc66c82f27cf9d5f. You have to remove that container to be able to reuse that name: that name is already in use by an external entity`

It is already wrapped in the error.